src/mail/model/MailModel.ts

```diff
@@ -398,8 +400,26 @@ export class MailModel {
 				_ownerGroup: mailboxGroupRoot._ownerGroup,
 			}))
 		}
-		return this.entityClient.load(MailboxPropertiesTypeRef, mailboxGroupRoot.mailboxProperties)
+		const mailboxProperties = await this.entityClient.load(MailboxPropertiesTypeRef, mailboxGroupRoot.mailboxProperties)
+		if (mailboxProperties.mailAddressProperties.length === 0) {
+			await this.migrateFromOldSenderName(mailboxGroupRoot, mailboxProperties)
+		}
+		return mailboxProperties
 	}
+	private async migrateFromOldSenderName(mailboxGroupRoot: MailboxGroupRoot, mailboxProperties: MailboxProperties) {
```

Comment:
i'd add a comment or amend the commit message to state that the migration entails that we're using the legacy name for all addresses of the user

Reply:
I will add a comment on what it means but otherwise it is in acceptance criteria